streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -315,20 +314,34 @@ public void shouldRecursivelyPrintChildren() throws Exception {
         assertThat(result, containsString("child-two:\n\t\tchildren:\t[child-two-one]"));
     }
 
+    @Test
+    public void shouldConsiderTimeStamps() throws Exception {
+        final int partition = 10;
+        driver = new ProcessorTopologyTestDriver(config, createSimpleTopology(partition).internalTopologyBuilder);
+        driver.process(INPUT_TOPIC_1, "key1", "value1", STRING_SERIALIZER, STRING_SERIALIZER, 10L);
+        driver.process(INPUT_TOPIC_1, "key2", "value2", STRING_SERIALIZER, STRING_SERIALIZER, 20L);
+        driver.process(INPUT_TOPIC_1, "key3", "value3", STRING_SERIALIZER, STRING_SERIALIZER, 30L);
+        assertNextOutputRecord(OUTPUT_TOPIC_1, "key1", "value1", partition, 10L);
+        assertNextOutputRecord(OUTPUT_TOPIC_1, "key2", "value2", partition, 20L);
+        assertNextOutputRecord(OUTPUT_TOPIC_1, "key3", "value3", partition, 30L);
+    }
+
+
     private void assertNextOutputRecord(String topic, String key, String value) {
-        ProducerRecord<String, String> record = driver.readOutput(topic, STRING_DESERIALIZER, STRING_DESERIALIZER);
-        assertEquals(topic, record.topic());
-        assertEquals(key, record.key());
-        assertEquals(value, record.value());
-        assertNull(record.partition());
+        assertNextOutputRecord(topic, key, value, null, 0L);
     }
 
     private void assertNextOutputRecord(String topic, String key, String value, Integer partition) {
+        assertNextOutputRecord(topic, key, value, partition, 0L);
+    }
+
+    private void assertNextOutputRecord(String topic, String key, String value, Integer partition, Long timestamp) {
```

Comment:
nit: make the parameters final, and if you could put them on separate lines. Ditto for `assertNextOutputRecord` on lines 330 and 334 above